build-android/app/src/main/java/com/google/firebase/codelab/friendlychat/SignInActivity.kt

```diff
@@ -45,61 +43,57 @@ class SignInActivity : AppCompatActivity() {
         // See: https://developer.android.com/topic/libraries/view-binding
         binding = ActivitySignInBinding.inflate(layoutInflater)
         setContentView(binding.root)
+    }
 
-        // Set click listeners
-        binding.signInButton.setOnClickListener { signIn() }
+    public override fun onStart() {
+        super.onStart()
 
-        // Configure Google Sign In
-        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
-            .requestIdToken(getString(R.string.default_web_client_id))
-            .requestEmail()
-            .build()
-        signInClient = GoogleSignIn.getClient(this, gso)
+        // If there is no signed in user, launch FirebaseUI
+        // Otherwise head to MainActivity
+        if (Firebase.auth.currentUser == null) {
+            // Sign in with FirebaseUI, see docs for more details:
+            // https://firebase.google.com/docs/auth/android/firebaseui
+            val signInIntent = AuthUI.getInstance()
+                    .createSignInIntentBuilder()
+                    .setLogo(R.mipmap.ic_launcher)
+                    .setAvailableProviders(listOf(
+                            AuthUI.IdpConfig.EmailBuilder().build(),
+                            AuthUI.IdpConfig.GoogleBuilder().build(),
+                    ))
+                    .build()
 
-        // Initialize FirebaseAuth
-        auth = Firebase.auth
-    }
-
-    private fun signIn() {
-        val signInIntent = signInClient.signInIntent
-        startActivityForResult(signInIntent, RC_SIGN_IN)
```

Comment:
Android Studio tells me this is deprecated, must have changed recently. Worth updating in this codelab?

Reply:
Yeah we'll be able to fix this when FirebaseUI 7.2.0 is released. I'll file a tracking issue here.

Reply:
https://github.com/firebase/codelab-friendlychat-android/issues/175